### PR TITLE
feat(rum): export rum types (#1045)

### PR DESCRIPTION
**Pull request by devcorpio**
# Summary

With this change the following types are going to be exposed:

- Transaction
- Span
- TransactionOptions
- SpanOptions

### CODE REVIEW

**Comment by devcorpio** [comment]
In the file ```packages/rum/src/index.d.ts``` we still have `declare class Transaction` and `declare class Span`. Since now we have the types we don't need those statements anymore.

Although if we remove them we could be forcing users that were already relying on them given typescript make them available anyway! (without import)

Should we remove them? Any thoughts?

**Comment by devcorpio** [comment]
Could we have any unexpected issue for removing` declare class SpanOptions` and `declare class Transaction`?

File: packages/rum/src/index.d.ts [hunk 1]
```diff
@@ -24,6 +24,51 @@
  */
 
 declare module '@elastic/apm-rum' {
+  export class Span extends BaseSpan {
+    sync: boolean
```
Comment (vigneshshanmugam) [inline packages/rum/src/index.d.ts:28]: nit: this is optional

File: packages/rum/src/index.d.ts [hunk 2]
```diff
@@ -24,6 +24,51 @@
...
+     * undocumented, might be removed in future versions
+     */
+    isFinished(): boolean
+  }
+  export interface SpanOptions {
+    /**
+     * undocumented, might be removed in future versions
+     */
+    startTime?: number
+    sync: boolean
```
Comment (vigneshshanmugam) [inline packages/rum/src/index.d.ts:59]: would make it as optional as well.

**Review by vigneshshanmugam — APPROVED** [review]
LGTM

**Pull request merged**